pkg/apis/application/v1alpha1/types.go

```diff
@@ -223,6 +223,9 @@ type Cluster struct {
 
 	// Config holds cluster information for connecting to a cluster
 	Config ClusterConfig `json:"config" protobuf:"bytes,3,opt,name=config"`
+
+	// Message can hold a status message or error.
+	Message string `json:"-" protobuf:"bytes,4,opt,name=message"`
```

Comment:
Just to add -- this is typically something that would typically belong under a `status` field of an object for normal k8s objects. But since this is just lightweight config objects, we don't have spec/status so everything is just going into a flatter definition.

Reply:
@jessesuen updated